### PR TITLE
eDSL doc refactor for dev kit

### PR DESCRIPTION
**Pull request by dgkutnic**

### CODE REVIEW

File: docs/usage/edsl_ops.rst [hunk 1]
```diff
@@ -429,6 +429,50 @@ Special Operations
...
+      :pyobject: layer
+
+- Trace
+
+While degugging eDSL code it can often be desirable to insert print statements to
+understand what the program is doing as it executes. However, typical print
+statements are not generally useful for this, as execution of eDSL occurs deep within
+the PlaidML backend. The ``trace`` operation allows print statements to be specified
+within the eDSL program and printed during the program's execution. Additionally, if
+multiple tracepoints are specified, the time ellapsed between consecutive tracepoints
```
Comment (flaub) [inline docs/usage/edsl_ops.rst:460]: Typo: elapased

File: docs/usage/edsl_ops.rst [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+      :end-before: layer_end
+  
+  .. group-tab:: Python
+
+    .. literalinclude:: ../../plaidml/edsl/edsl_docs.py
+      :pyobject: layer
+
+- Trace
+
+While degugging eDSL code it can often be desirable to insert print statements to
```
Comment (dgkutnic) [inline docs/usage/edsl_ops.rst:455]: typo: debugging

File: docs/usage/edsl_examples.rst [hunk 3]
```diff
@@ -314,15 +326,270 @@ coefficients, and ``P`` gives the padding offsets.
...
+    &{\color{red}\verb!.outAccess(i,j)!}\\
+    &{\color{green}\verb!.sum(!}\\
+    &{\color{blue}\verb!        A(!}
+    \verb!i,k!
+    {\color{blue}\verb!)!}
+    {\color{orange}\verb!  *  !}
+    {\color{blue}\verb!B(!}
+    \verb!k,j!
+    {\color{blue}\verb!)!}\\
+    &{\color{green}\verb!     )!}\\
```
Comment (tzerrell) [inline docs/usage/edsl_examples.rst:514]: To my eye, at least, the interior of the sum and the final close `)` are indented ~1 tab too far. Perhaps remove the leading spaces?

File: docs/usage/edsl_examples.rst [hunk 4]
```diff
@@ -314,15 +326,270 @@ coefficients, and ``P`` gives the padding offsets.
...
+
+    .. literalinclude:: ../../plaidml/edsl/examples/gemm.py
+      :pyobject: gemm
+
+
+Uniform 8-bit Quantization
+*******************
+
+Quantization is a promising new approach used to reduce memory and compute resources required by neural network operations.
+This examples illustrates 8-bit uniform quantization operation :cite:`ob2018quantization` in eDSL.
```
Comment (tzerrell) [inline docs/usage/edsl_examples.rst:543]: The citation here is not rendering correctly on my machine, but perhaps I'm building it wrong.

**Review by tzerrell — COMMENTED** [review]
@1sanz The content of these docs looks generally good to me; I have a few minor comments

File: docs/usage/edsl_examples.rst (hunk 4 above)
Comment (1sanz) [reply docs/usage/edsl_examples.rst]: Thanks Tim! There was a typo. I have fixed it and also changed the style to a cleaner numbered one. See if this seems better.

File: plaidml/edsl/edsl_docs.py [hunk 5]
```diff
@@ -37,8 +37,8 @@ def matmul(A, B):
 def global_min(I):
     i, j, k = TensorIndexes(3)
     Neg = -I
-    O = Contraction().max(Neg[i, j, k])
-    return -O
+    O = -Contraction().max(Neg[i, j, k])
```
Comment (tzerrell) [inline plaidml/edsl/edsl_docs.py:40]: Can you make this same change in the C++ version of this code?

File: docs/usage/edsl_examples.rst (hunk 4 above)
Comment (tzerrell) [reply docs/usage/edsl_examples.rst]: Looks good now, thanks for fixing!

File: plaidml/edsl/edsl_docs.py (hunk 5 above)
Comment (tzerrell) [reply plaidml/edsl/edsl_docs.py:40]: Thanks!

File: plaidml/edsl/edsl_docs.py [hunk 6]
```diff
@@ -15,44 +15,38 @@ def sum_over_axis(I):
...
 
 def global_min(I):
     i, j, k = TensorIndexes(3)
     Neg = -I
-    O_Neg = TensorOutput()
-    O_Neg[()] >= Neg[i, j, k]
-    O = -O_Neg
-    return O
+    O_Neg = Contraction().max(Neg[i, j, k])
+    return -O_Neg
```
Comment (tzerrell) [inline plaidml/edsl/edsl_docs.py:41]: Does it turn out that we do need the separate `O_Neg` line for this to work correctly?

**Review by tzerrell — COMMENTED** [review]
Looking good, one inline comment

File: plaidml/edsl/edsl_docs.py (hunk 6 above)
Comment (1sanz) [reply plaidml/edsl/edsl_docs.py:41]: There was a small issue with cpp. Please see the update.

Comment (tzerrell) [reply plaidml/edsl/edsl_docs.py]: great, looks good!

**Review by tzerrell — APPROVED** [review]

**Review by tzerrell — APPROVED** [review]

**Pull request merged**